### DISCUSSION TITLE
What are the differences in git repository settings?

### DISCUSSION THREAD

**Discussion by josh-nz** [glushchenko/fsnotes#1796] [General]
There are two options on the Git settings page:

- Custom worktree, or
- Separate .git in project dir (except iCloud Drive)

I don't understand the differences between these two options. 

Can someone please help explain? Thanks.

**Comment by glushchenko** [comment]
Hi @josh-nz, 

With this setting you can store the repository in a folder you specify, or in the folder where your documents are located.

**Reply by josh-nz** [reply]
I'm still confused as to which I should use and when. Can you explain the differences in a little more detail?

Of what benefit is a worktree? A worktree still has the `.git` folder present in it, and the feature was designed for working in multiple branches at once, something that doesn't seem to be a need for FSNotes.

The worktree also needs to have a branch, so what do you name this branch?

The option for a `.git` folder in the project dir doesn't seem to be allowed for the ICloud Drive folder, but I don't see how git worktrees solves this problem since the `.git` folder is still present in the worktree. Can you not use the git option at all if your notes are being saved to the iCloud Drive folder?

Thanks

**Reply by glushchenko** [reply]
Where to store the .git repository is a matter of personal preference. I don't have any recommendations on this. The default is Custom worktree. The worktree function is used to change the path to the repository and nothing more.
